### ISSUE TITLE
Suspected false positives in v0.4.26

### ISSUE THREAD

**Issue by baraknaveh** [ArnaudBarre/eslint-plugin-react-refresh#100]
I'm getting:
`Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.` 

in this example:

```Typescript
import { cn } from "@/lib/utils";

// Exported for drop area, MenuButton
export const ACTION_BUTTON_STYLE =  ///// <--  flagged with error
  "flex rounded-full py-2 px-3 items-center justify-center";
export const ACTION_BUTTON_PRIMARY_STYLE = cn(
  /* normal.. */ "bg-zinc-900 text-white",
  /* hover... */ "hover:bg-zinc-700",
  /* active.. */ "active:bg-zinc-500",
  /* disabled */ "disabled:bg-zinc-100 disabled:text-zinc-400",
);
```
where `@/lib/utils.ts` is:
```typescript
import { clsx, type ClassValue } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}
```

Why am I getting the report? is this not a false positive?

**Comment by baraknaveh** [comment]
Many thanks!

**Comment by baraknaveh** [comment]
I tried `0.5.0-alpha.0` and I'm now getting the same error but on other constants like:

```typescript
const SignIn = lazy(() =>
  import("./routes/SignUpSignIn").then((m) => ({ default: m.SignIn })),
);
```

or 
```typescript
export const GoogleDocMimeType =
  "application/vnd.google-apps.document" as const;
```

Are these false positives? or is this by design?

Version `0.5.0-alpha.0` flags only 3 files for me, so I can disable the check for them specifically, but if it's a real issue, I'd rather fix by refactoring. Seem like false positives though.

**Comment by ArnaudBarre** [comment]
Can you provide more context for both reports?
In the first case, are you exporting something that is not a React component in the same file?
  - If yes, this is a valid report
  - If no, this an issue in the rule and a small repro would help fixing this

In the second case, do you have React component in the file?
  - If yes, exporting a string is allow only if the `allowConstantExport` option is enabled. If you are using Vite, you can enable it, otherwise splitting is best to avoid hard refresh in Next
  - If no, this an issue in the rule and a small repro would help fixing this. If the file doesn't include jsx, you can rename it to `.ts`,  but I'm still curious about having a repro if you can
